### PR TITLE
Issue/10444 add tracks to unpublished revision dialog

### LINKED ISSUE #10444: Add Tracks to the new Unpublished Revision dialog
I think it'd be nice to track the following
- dialog shown ("unpublished_revision_dialog_shown")
- button clicks ("unpublished_revision_dialog_load_local_version_clicked", "unpublished_revision_dialog_load_unpublished_version_clicked")

cc @shiki Can you please make sure the same events are tracked in iOS. Feel free to update the names of the events, I just used the first names that came to my mind. Thanks!

### PR DESCRIPTION
**Pull request by malinajirka**
Fixes #10444 

Adds tracking events to unpublished revision dialog.

To test:

1. Create an unpublished revision in Calypso
2. Open Post List in the app
3. Notice the "Unhandled auto save" label
4. Click on the post -> Make sure `unpublished_revision_dialog_shown` is shown in LogCat
5. Choose "More recent version/The version from another device" -> Make sure `unpublished_revision_dialog_load_unpublished_version_clicked` is shown in LogCat
6. Click back (don't change the post)
7. Click on the post again
8. Choose "Current version/The version from this device" -> Make sure `unpublished_revision_dialog_load_local_version_clicked` is shown in LogCat

Update release notes:

- [ x ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by jkmassel** [comment]
Howdy! The `13.3` code freeze is today, so I'm bumping this to `13.4`. If you'd like it to be part of the `13.3` release, please target the `release/13.3` branch, and DM me once it's merged so that I can include it as a new beta release!

File: WordPress/src/main/java/org/wordpress/android/ui/posts/PostListDialogHelper.kt [hunk 1]
```diff
@@ -18,7 +22,8 @@ private const val CONFIRM_ON_AUTOSAVE_REVISION_DIALOG_TAG = "CONFIRM_ON_AUTOSAVE
  */
 class PostListDialogHelper(
     private val showDialog: (DialogHolder) -> Unit,
-    private val checkNetworkConnection: () -> Boolean
+    private val checkNetworkConnection: () -> Boolean,
+    private val analyticsTracker: AnalyticsTrackerWrapper
```
Comment (oguzkocer) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/PostListDialogHelper.kt]: I am not a big fan of passing the `AnalyticsTrackerWrapper` to the `PostListDialogHelper` and I'd have personally preferred to move the positive dialog actions to their respective methods and handling the show dialog tracking in a different way, but I can't justify requesting such a change after the fact as I don't think there would be significant enough benefit to it. So, I am just leaving a note with my thoughts, but you should basically ignore this :)

Just for the sake of sharing, I was thinking of adding a `PostListDialogType` `enum` with tag and tracking info attached to it and passing that back to the main view model in `showDialog`, but yeah, not really worth it and might even end up being confusing 🤷‍♂

Comment (malinajirka) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/PostListDialogHelper.kt]: Thanks for sharing! ;)

I think the solution you are proposing would be nice, but at the same time I agree with you that it's probably not worth the changes. I also feel a bit that both logging and tracking aren't real dependencies -> I don't mind, that much, if they are used even in classes where they "don't belong".

**Comment by shiki** [comment]
Thank you, @oguzkocer!

**Pull request merged**